### ISSUE TITLE
incorrect error message: The class couldn't be fully loaded...

### ISSUE THREAD

**Issue by lszl84** [godotengine/godot#37008]
**Godot version:** 3.2.1 stable

**OS/device including version:** Mac OS 10.15.3

**Issue description:** Incorrect error message: `The class "SingleItemClickable" couldn't be fully loaded...`. There's no reason for such error, and game works without any problems

**Steps to reproduce:** Open attached project and open BaseLevel.gd. The error appears at the end of the file. 

**Minimal reproduction project:**
[Lake copy.zip](https://github.com/godotengine/godot/files/4324756/Lake.copy.zip)

Tried to strip the project to bare minimum, that's why it looks weird. I suppose the problem may be with autoloads, so I left the classes there.

This might be a duplicate of https://github.com/godotengine/godot/issues/21461 , but here you have a minimal product that reproduces the bug every time (at least in my case).

**Comment by Ev1lbl0w** [comment]
That's because you indeed have an error on **SingleItemClickable.gd**. It looks like it can't find `ItemDb.ID`. I assume it's because `ItemDb` is an autoload, and in the current context *(when exporting a variable)*, it doesn't exist.

If you add `tool` at the top, now it works.

**Comment by lszl84** [comment]
So in editor context, the autoload scripts are not loaded? Adding `tool` at the beginning of a script forces it to be run in the editor and the editor loads autoload objects -- all of them, or just the ones used by the script? Why the editor can't load the autoload script if it's obviously marked as the type of the exported variable?

BTW adding `tool` (and also restarting Godot after that) didn't help:

<img width="715" alt="Screenshot 2020-03-12 at 19 06 44" src="https://user-images.githubusercontent.com/42072702/76553649-b7371800-6494-11ea-8fd1-98e22f05deeb.png">

**Comment by Ev1lbl0w** [comment]
You're right, it seems that this bug is very inconsistent. I just *retried* my steps and doesn't work. However, after tinkering for it like 5 minutes, suddenly it recognizes `ItemDb.ID`. And it doesn't even need to be a tool to work. So, weird indeed.

I'll try looking at the code, but can't promise much.
